sdk/ml/azure-ai-ml/tests/internal_utils/unittests/test_ml_client.py

```diff
@@ -606,3 +606,101 @@ def test_enable_telemetry(self) -> None:
             )
             assert not enable_telemetry
             assert isinstance(handler, logging.NullHandler)
+
+    @pytest.mark.parametrize(
+        "args, kwargs, ops_name, create_method_name",
+        [
+            (
+                [load_compute("tests/test_configs/compute/compute-ci.yaml")],
+                {},
+                "compute",
+                "begin_create_or_update",
+            ),
+            (
+                [load_online_endpoint("tests/test_configs/endpoints/online/online_endpoint_create_k8s.yml")],
+                {},
+                "online_endpoints",
+                "begin_create_or_update",
+            ),
+            (
+                [load_online_deployment("tests/test_configs/deployments/online/online_deployment_blue.yaml")],
+                {},
+                "online_deployments",
+                "begin_create_or_update",
+            ),
+            (
+                [load_batch_endpoint("tests/test_configs/endpoints/batch/batch_endpoint_mlflow.yaml")],
+                {},
+                "batch_endpoints",
+                "begin_create_or_update",
+            ),
+            (
+                [load_batch_deployment("tests/test_configs/deployments/batch/batch_deployment_1.yaml")],
+                {},
+                "batch_deployments",
+                "begin_create_or_update",
+            ),
+        ],
+    )
+    def test_ml_client_begin_create_without_ws_or_registry_name(
+        self, args, kwargs, ops_name, create_method_name, mock_credential
+    ) -> None:
+        ml_client = MLClient(
+            credential=mock_credential,
+            subscription_id=Test_Subscription,
+            resource_group_name=Test_Resource_Group,
+            registry_name=None,
+        )
+
+        with pytest.raises((ValidationException, Exception)) as val_exception:
```

Comment:
Prefer catching the most specific Exception appropriate for the test

```suggestion
        with pytest.raises(ValidationException) as val_exception:
```

Reply:
I created a bug report for this. The ValidationException that is raised through the decorator to validate ws/registry name is passed through log_and_raise_error which is raising it as a regular Exception for some of the operations.

For reference:
https://dev.azure.com/msdata/Vienna/_workitems/edit/2283939/